### PR TITLE
refactor resources API

### LINKED ISSUE #1722: Resource API returns duplicate data
**Describe the bug(描述下问题)**
Resource API returns duplicate data when having the same creation time.

**Versions used(KubeSphere/Kubernetes的版本)**
KubeSphere: v2.1.0
Kubernetes: (If KubeSphere installer used, you can skip this)

image

image

https://kubesphere.com.cn/forum/d/383-bug

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Sort resources by name when the creation time is the same.

**Which issue(s) this PR fixes**:

Fixes #1722

### CODE REVIEW

File: pkg/models/resources/resources.go [hunk 1]
```diff
@@ -160,13 +162,18 @@ func ListResources(namespace, resource string, conditions *params.Conditions, or
...
-		if i >= offset && (limit == -1 || len(items) < limit) {
-			items = append(items, injector.addExtraAnnotations(item))
-		}
+	totalCount := len(result)
+
+	if limit == -1 || limit+offset > totalCount {
+		limit = totalCount - offset
+	}
+
+	result = result[offset : offset+limit]
```
Comment (zryfish) [inline pkg/models/resources/resources.go]: what if offset > totalCount?

File: pkg/server/params/params.go [hunk 2]
```diff
@@ -45,7 +44,9 @@ func ParsePaging(paging string) (limit, offset int) {
 	return
```
Comment (zryfish) [inline pkg/server/params/params.go]: why ignore error here? what would happen if there is an error during parsing?

Comment (wansir) [reply pkg/server/params/params.go]: Use default values when parsing failed, limit and page must be defined at the same time.

Comment (zryfish) [reply pkg/server/params/params.go]: I believe atoi will return 0 when there is an error

Comment (wansir) [reply pkg/server/params/params.go]: Regular expression `\d+` defines that only numbers are allowed here, i think this error can be ignored.

**Pull request merged**